spec/support/feature_helpers/navigation.rb

```diff
@@ -1,10 +1,13 @@
 module FeatureHelpers
   module Navigation
-    def and_i_should_be_on page
+    def and_i_should_be_on page, sponsor_name = ''
+      sponsor = Sponsor.find_by(name: sponsor_name) if not sponsor_name.empty?
       page_path = case page.to_sym
                     when :root_page then hq_root_path
                     when :sign_in_page then new_hq_admin_user_session_path
                     when :new_sponsor_page then new_hq_sponsor_path
+                    when :hq_sponsor_path then hq_sponsor_path sponsor.id
+                    when :hq_new_sponsorship_path then hq_new_sponsorship_path sponsor.id
                     else raise('path to specified is not listed in #path_to')
                   end
```

Comment:
This is not very reusable. If we need it to use it for `orpahns_name` it will be a problem.
Let's refactor this to respect the SOLID design principles.


One option would be to rename this `def and_i_should_be_on path` to  `def get_path path, id`

Then to handle the conversion from name to id:
``` ruby
#use like:  and_i_should_be_on "hq_sponsor_path", sponsor_name: "Sponsor 1"
def and_i_should_be_on path, field_hash
     case  field_hash.keys.first.to_sym
          when :sponsor_name do
                 sponsor = Sponsor.find_by(name: field_hash[:sponsor_name])
                 get_path path, sponsor.id
         end
          when :sponsor_id do
                 get_path path, field_hash[:sponsor_id]
         end
         else raise('some error message')
   end

end
```
(this wasn't tested, i got carried away, refactor where needed)